src/consensus/mod.rs

```diff
@@ -0,0 +1,56 @@
+use futures::{FutureExt, Stream, StreamExt};
+
+use crate::transport{Endpoint, Client, Request, Response, proto::{self}};
+
+pub struct FastPaxos<'a, C> {
+    client: &'a C,
+    my_addr: Endpoint,
+    size: usize
+}
+
+impl<'a, C> FastPaxos<'a, C> {
+    pub fn new<C>(my_addr: Endpoint, size: usize, client: &mut C) -> FastPaxos
+    where
+        C: Client
+    {
+        FastPaxos {
+            my_addr, size, client
+        }
+    }
+
+    pub async fn propose(&self, proposal: &[Endpoint]) {
+        // TODO: register proposal with regular paxos instance
+        unimplemented!();
+    }
+
+    pub async fn handle_message(&self, request: Request) -> crate::Result<Response> {
+        match request.kind {
```

Comment:
I think what we should do here is maybe make all the paxos requests a sub enum? This would allow us to match them at the Membership level then avoid this error.

Reply:
I moved it to it's own enum, but I still need to take the request as a parameter, so I have to have the exhaustive match. I mean you could pass the enum and the request separately, but that kinda defeats the purpose of the `Request` type. I'm guessing this is what structural typing means..